forge-game/src/main/java/forge/game/ability/effects/AnimateEffectBase.java

```diff
@@ -158,6 +158,7 @@ public static void doAnimate(final Card c, final SpellAbility sa, final Integer
         final List<Trigger> addedTriggers = Lists.newArrayList();
         for (final String s : triggers) {
             final Trigger parsedTrigger = TriggerHandler.parseTrigger(AbilityUtils.getSVar(sa, s), c, false, sa);
+            if (perpetual) parsedTrigger.setSpawningAbility(sa);
```

Comment:
With this line the previously broken EOT triggers on Goblin Trapfinder and Flames of Moradin will trigger when expected...

BUT the Sacrifice always whiffs. Debugging in SacrificeEffect shows that the zone of the card is the Stack somehow. The GUI still shows the card on the Battlefield.

Does the trigger going to the stack somehow move the host card of the trigger to the stack as well??

Reply:
I don't think you can do that (making them look like delayed triggers to the engine)
they're gonna trigger from any zone now...

Reply:
OK, back to the drawing board

Reply:
Probably the delayed trigger thing is the reason the engine assumed the source was not on the battlefield anyway